foreigndonors.php

```diff
@@ -195,21 +196,44 @@ function foreigndonors_civicrm_buildForm($formName, &$form) {
     $form->assign('domainId', $domainId);
     $form->assign('foreignPageId', $formId);
 
+    // As a safeguard against people completing NSW contrib forms through the wrong domain
+    // We're going to get the financial type of the page/event and if it starts with 'NSW'
+    // we know to inject the prohibited donor checkbox, regardless of domain
+    $finType = [];
+    if ($formName == 'CRM_Contribute_Form_Contribution_Main') {
+      $finType = \Civi\Api4\ContributionPage::get(FALSE)
+        ->addSelect('financial_type_id:label')
+        ->addWhere('id', '=', $formId)
+        ->setLimit(1)
+        ->execute();
+    }
+    else {
```

Comment:
I'd check here to see if it's an Event.  I mean, theoretically it should be, but safest would be to check.

Reply:
We're inside a code block that only executes if the page is a contrib page or event registration page already. It can literally be only one of two values by the time we reach this part of the code.